### PR TITLE
[Enhancement] Support resize or rescale to multiple

### PR DESCRIPTION
**Pull request by Junjun2016**
## Motivation
To meet the requirement of some networks that the size of the input image should be multiple to the divisor

## Modification
Add `divisor` argument to `imresize` and `imrescale` in `mmcv.image`.
This argument rounds up the resized or rescaled image size to the nearest value that can be divisible by the divisor.

### CODE REVIEW

File: mmcv/image/geometric.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 import cv2
 import numpy as np
+from torch.nn.modules.utils import _pair as to_2tuple
```
Comment (ZwwWayne) [inline mmcv/image/geometric.py:6]: try not to add it here in case parrots do not support this internal API.

Comment (xvjiarui) [reply mmcv/image/geometric.py:6]: How about we implement it in MMCV?

Comment (ZwwWayne) [reply mmcv/image/geometric.py]: > How about we implement it in MMCV?

agree

File: mmcv/image/geometric.py [hunk 2]
```diff
@@ -92,6 +94,71 @@ def imresize(img,
...
+                         size=None,
+                         scale_factor=None,
+                         keep_ratio=False,
+                         return_scale=False,
+                         interpolation='bilinear',
+                         out=None,
+                         backend=None):
+    """Resize image according to a given size or scale factor and then rounds
+    up the the resized or rescaled image size to the nearest value that can be
+    divisible by the divisor.
```
Comment (ZwwWayne) [inline mmcv/image/geometric.py:108]: can be divided

File: mmcv/image/geometric.py [hunk 3]
```diff
@@ -92,6 +94,71 @@ def imresize(img,
...
+
+    Args:
+        img (ndarray): The input image.
+        divisor (int | tuple): Resized image size will be multiple to divisor.
```
Comment (ZwwWayne) [inline mmcv/image/geometric.py:112]: be a multiple of ``divisor``

File: mmcv/image/geometric.py [hunk 4]
```diff
@@ -92,6 +94,71 @@ def imresize(img,
...
+            If divisor is tuple, divisor is (w_divisor, h_divisor).
```
Comment (ZwwWayne) [inline mmcv/image/geometric.py:113]: is a tuple, divisor should be

File: mmcv/image/geometric.py [hunk 5]
```diff
@@ -92,6 +94,71 @@ def imresize(img,
...
+        size (None | int | tuple[int]): Target size (w, h). Default: None.
+        scale_factor (None | float | tuple[float]): Multiplier for spatial
+            size. Has to match input size if it is a tuple and the 2D style is
```
Comment (ZwwWayne) [inline mmcv/image/geometric.py:116]: Should match input size

File: mmcv/image/geometric.py (hunk 1 above)
Comment (Junjun2016) [reply mmcv/image/geometric.py]: Refer to this [PR](https://github.com/open-mmlab/mmcv/pull/1125)

**Review by ZwwWayne — APPROVED** [review]

**Comment by ZwwWayne** [comment]
LGTM, can be merged if @xvjiarui has no further comments.

File: mmcv/image/geometric.py [hunk 6]
```diff
@@ -92,6 +94,72 @@ def imresize(img,
...
+    h, w = img.shape[:2]
+    if size is not None and scale_factor is not None:
+        raise ValueError('only one of size or scale_factor should be defined')
+    elif size is None and scale_factor is None:
+        raise ValueError('one of size or scale_factor should be defined')
+    elif size is not None:
+        if isinstance(size, int):
+            size = (size, size)
+        if keep_ratio:
+            size = rescale_size((w, h), size, return_scale=False)
```
Comment (xvjiarui) [inline mmcv/image/geometric.py:143]: If user input a float number, no error will be raised. But `rescale_size` has different effect for float and tuple[int]. 
We may check `isinstance(size, (int, float))`.

**Pull request merged**